patch_migrations.py

```diff
@@ -0,0 +1,154 @@
+	'address',
+	'transactionHash',
+	# 'transactionIndex', transaction index needs a different fix
+	'blockHash'
+]
+
+columns = {
+    'blocks': [blocks_columns, ('number')],
+    'withdrawals': [withdrawals_columns, ('block', 'wtdrlIndex')],
+    'transactions': [transactions_columns, ('id')],
+    'event_logs': [event_logs_colums, ('block', 'logIndex')],
+}
+
+
+class DatabaseManager:
+    def __init__(self, db_name, table_name):
+        self.db = db_name
+        self.table = table_name
+
+    @contextmanager
+    def connect(self):
+        try:
+            conn = sqlite3.connect(self.db)
+            yield conn
+        except:
+            raise
+        finally:
+            conn.close()
+
+    def execute(self, query):
+        with self.connect() as conn:
+            cursor = conn.cursor()
+            cursor.execute(query, params or ())
+            conn.commit()
+            return cursor.rowcount
+
+    def fetchall(self, query, params=None):
+        with self.connect() as conn:
+            cursor = conn.cursor()
+            cursor.execute(query, params or ())
+            return cursor.fetchall()
+
+    def fetchone(self, query, params=None):
+        with self.connect() as conn:
+            cursor = conn.cursor()
+            cursor.execute(query, params or ())
+            return cursor.fetchone()
+
+class DataManipulator:
+    def __init__(self, connection, table):
+        self.conn = connection
+        self.tbl = table
+        self.prmry = columns.get(table)[1]
+        self.initial = {}
+
+    def get_initial_results(self):
+        primary = self.prmry
+        table = self.tbl
+
+        if isinstance(primary, str):
+            partial = f', {primary} FROM {table};'
+        elif isinstance(primary, tuple):
+            partial = f', {primary[0]}, {primary[1]} FROM {table};'
+
+        for column in columns.get(table)[0]:
+            q = f'SELECT hex({column})' + partial
+            rows = self.conn.fetchall(q)
+            zeros = find_zeros(rows)
+            prepared = remove_zeros(zeros)
+            
+            if len(prepared) > 0:
+                self.initial[column] = prepared
+
+        return self.initial
+
+    def alter_table(self):
+        initial_results = self.get_initial_results()
+
+        primary = self.prmry
+        table = self.tbl
+
+        for k, v in initial_results.items():
+            print(f"processing column: {k} on table: {table} with {len(v)} statements to process")
+            for item in v:
+                if isinstance(primary, str):
+                    s = f'UPDATE {table} SET {k} = {item[0]} WHERE {primary} = {item[1]}'
+                elif isinstance(primary, tuple):
+                    s = f'UPDATE {table} SET {k} = {item[0]} WHERE {primary[0]} = {item[1]} AND {primary[1]} = {item[2]}'
+                print(s)
+                # self.conn.execute(s)
+                # I have the above commented and have added the print call for testing / inspecting
+
```

Comment:
I don't think the following functions really do what we want.

We want to remove 0 bytes, but the query is getting hex values, so that's what we're working with. In hex, a zero byte is '00', not '0'. This would catch values beginning with things like '01' and '0a' which are not zero bytes and should not be trimmed, and if you shorten '0a' to 'a' then your hex string will be an odd length, which the hex decoder in SQLITE will reject as an invalid input *if* you're indicating that the value is a hex string, which I don't think you are.

Now, I'm not sure how Python's sqlite library will work if you queried for raw values instead of hex values. If that works, I'd do that. If it doesn't, I'd get the hex values, use the codecs library to decode them, then do find_zeros and remove_zeros on the byte strings (in which case you will compare to `0` instead of `'0'`).